packages/driver/src/multi-domain/communicator.ts

```diff
@@ -19,47 +19,41 @@ const CROSS_DOMAIN_PREFIX = 'cross:domain:'
  * @extends EventEmitter
  */
 export class PrimaryDomainCommunicator extends EventEmitter {
-  private windowReference: Window | undefined
   private crossDomainDriverWindows: {[key: string]: Window} = {}
   userInvocationStack?: string
 
   /**
-   * Initializes the event handler to receive messages from the spec bridge.
-   * @param {Window} win - a reference to the window object in the primary domain.
+   * The callback handler that receives messages from secondary domains.
+   * @param {MessageEvent.data} data - a reference to the MessageEvent.data sent through the postMessage event. See https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/data
+   * @param {MessageEvent.source} source - a reference to the MessageEvent.source sent through the postMessage event. See https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/source
    * @returns {Void}
    */
-  initialize (win: Window) {
-    if (this.windowReference) return
-
-    this.windowReference = win
-
-    this.windowReference.top?.addEventListener('message', ({ data, source }) => {
-      // currently used for tests, can be removed later
-      if (data?.actual) return
-
-      // check if message is cross domain and if so, feed the message into
-      // the cross domain bus with args and strip prefix
-      if (data?.event?.includes(CROSS_DOMAIN_PREFIX)) {
-        const messageName = data.event.replace(CROSS_DOMAIN_PREFIX, '')
-
-        // NOTE: need a special case here for 'bridge:ready'
-        // where we need to set the crossDomainDriverWindow to source to
-        // communicate back to the iframe
-        if (messageName === 'bridge:ready' && source) {
-          this.crossDomainDriverWindows[data.domain] = source as Window
-        }
+  onMessage ({ data, source }) {
+    // currently used for tests, can be removed later
+    if (data?.actual) return
```

Comment:
This can be removed now.

Reply:
awesome. will push a commit that gets rid of this!